internal/transformer/classes/generators/dockerfilegenerators/utils.go

```diff
@@ -17,15 +17,108 @@
 package dockerfilegenerators
 
 import (
+	"bufio"
+	"fmt"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strings"
 
 	"github.com/konveyor/move2kube/environment"
 	"github.com/konveyor/move2kube/internal/common"
+	"github.com/konveyor/move2kube/types/source/dotnet"
 	transformertypes "github.com/konveyor/move2kube/types/transformer"
 	"github.com/konveyor/move2kube/types/transformer/artifacts"
 	"github.com/sirupsen/logrus"
 )
 
+// isSilverlight tests whether the app is a sliverlight web application
+func isSilverlight(configuration dotnet.CSProj) (bool, error) {
+	if configuration.ItemGroups == nil || len(configuration.ItemGroups) == 0 {
+		return false, fmt.Errorf("No item groups in project file to parse")
+	}
+
+	for _, ig := range configuration.ItemGroups {
+		if ig.Contents == nil || len(ig.Contents) == 0 {
+			continue
+		}
+
+		for _, r := range ig.Contents {
+			if dotnet.WebSLLib.MatchString(r.Include) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// isWeb tests whether the project is a web application
+func isWeb(configuration dotnet.CSProj) (bool, error) {
+	if configuration.ItemGroups == nil || len(configuration.ItemGroups) == 0 {
+		return false, fmt.Errorf("No item groups in project file to parse")
+	}
+
+	for _, ig := range configuration.ItemGroups {
+		if ig.References == nil || len(ig.References) == 0 {
+			continue
+		}
+
+		for _, r := range ig.References {
+			if dotnet.WebLib.MatchString(r.Include) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// parseSolutionFile parses the solution file for the project file path
+func parseSolutionFile(inputPath string) []string {
+	solFile, err := os.Open(inputPath)
+	if err != nil {
+		logrus.Errorf("Could not open the solution file: %s", err)
+		return nil
+	}
+	defer solFile.Close()
+
+	solFileScanner := bufio.NewScanner(solFile)
+
+	r := regexp.MustCompile(dotnet.ProjBlock)
+
+	csr := regexp.MustCompile(dotnet.CsProj)
```

Comment:
If the solution file is an XML, why not use a XML parser?

Reply:
@ashokponkumar  solution files are text files, not xml...they have their own unique format .

Reply:
Okay sure. @pabloloyola do we know if there is a antlr4 parser for visual studio solution file